### DISCUSSION TITLE
Generate edit project in current folder

### DISCUSSION THREAD

**Discussion by galderz** [jbangdev/jbang#1282] [Q&A]
I was wondering `jbang edit` could generate the projects directly inside the folder from which is called from, rather than `{HOME}/.jbang`. I'm asking this because I'm often running jbang scripts in a different server, and these scripts are in a folder that is shared with my laptop via `sshfs`. Importantly, my entire home folder is not shared.

I have tried to share `{HOME}/.jbang` via `sshfs`, additionally from the other folder than I'm currently sharing, but that doesn't seem to work.

I looked around and don't see how you can tweak the path to the `.jbang` folder with `edit` command. Either allowing to tweak the location of the `.jbang` for the generated project, or have an option to generate the project inside the `.jbang` folder in the directory in which `jbang edit` is called would be fine for me.

**Comment by quintesse** [comment]
> have an option to generate the project inside the .jbang folder in the directory in which jbang edit is called

That's actually not a bad idea 👍 

Btw, there's this related issue https://github.com/jbangdev/jbang/issues/462 and this in-progress PR https://github.com/jbangdev/jbang/pull/1042 that are about creating the project in the same folder, which is not the same as your idea but you might be interested to give your opinion on that.

**Reply by galderz** [reply]
> > have an option to generate the project inside the .jbang folder in the directory in which jbang edit is called
> 
> That's actually not a bad idea 👍
> 
> Btw, there's this related issue #462 and this in-progress PR #1042 that are about creating the project in the same folder, which is not the same as your idea but you might be interested to give your opinion on that.

I think that'd work for me. The point is being able to control where the source code along with IDE supporting files stays when editing.

**Reply by maxandersen** [reply]
yeah - so could probably fairly easily allow to override the project dir; with the caveat it wont be cleaned up as it then will be outside the cache. can you open issue on it, please?

btw. @galderz if you are using intellij have you given the jbang plugin a try ? it doesn't require any of the generic magic we do in jbang edit today.

**Reply by quintesse** [reply]
> so could probably fairly easily allow to override the project dir

I have been thinking about this and I've come up with a couple of conflicting requirements:

- Being able to override the directory where the project/temp folder will be created (eg. specifying `--workspace=.jbang/projects` will create a folder with a generated name inside that folder)
- Being able to explicitly specify the folder where the project files will go (eg. one can specify `jbang edit --workspace=myproject myproject.java`. But I'm not sure how useful that is, especially given the fact that it means having to deal with wrong invocations like `jbang edit --workspace=projectA projectB.java`)
- Being able to edit in-place. (eg. specifying `--workspace=.`)

In light of that I wondered if it wouldn't be better to just have a simply option with a single function, eg specifying `--local` would mean setting the projects cache dir to `./.jbang/cache/projects` and nothing more. It would do exactly what @galderz wants , which seems very useful, and nothing more, no way to shoot oneself in the foot so to speak.

**Reply by maxandersen** [reply]
Problem is gradle search its parent dir for other build.gradle so I think if we put it directly inside and you have a build.gradle it won't work.

**Reply by quintesse** [reply]
You mean in case the parent dir where the user is writing the code happens to have a build.gradle file? Well we could check for that and warn the user that that won't work, right?

**Reply by galderz** [reply]
> btw. @galderz if you are using intellij have you given the jbang plugin a try ? it doesn't require any of the generic magic we do in jbang edit today.

I haven't yet. I'll install it and give it a go.
